compiler-rt/lib/mpk_untrusted/tests/unit/alloc_site_handler_tests.cpp

```diff
@@ -69,9 +72,81 @@ TEST(getAllocSite, ValidBetweenAddress) {
 }
 
 TEST(getAllocSite, PointerArithmeticOverflowWraps) {
-  auto ptr = (rust_ptr)(-1); // 0xffffffffffffffff
+  auto ptr = (rust_ptr)(-1); // max pointer size
   auto newAllocSite = std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1);
   EXPECT_FALSE(newAllocSite->containsPtr(++ptr));
 }
 
+TEST(faultingAllocs, addFaultAlloc) {
+  auto handle = AllocSiteHandler::getOrInit();
+  auto ptr = (rust_ptr)malloc(sizeof(uint64_t));
+  auto newAllocSite = std::make_shared<AllocSite>(ptr, sizeof(uint64_t), 1);
+  handle->insertAllocSite(ptr, newAllocSite);
+  handle->addFaultAlloc(ptr, 1);
+  auto fault_set = handle->faultingAllocs();
+  if (fault_set.empty())
+    FAIL() << "Error Adding AllocSite to error set!\n";
+
+  AllocSite ac = *fault_set.begin();
+  EXPECT_EQ(ac.getPtr(), newAllocSite->getPtr());
+  EXPECT_EQ(ac.id(), newAllocSite->id());
+  handle->removeAllocSite(ptr);
+}
+
+void *ThreadedGettid(void *tid) {
+  *(pid_t *)tid = gettid();
+  pthread_exit(NULL);
+}
+
+// Test gettid() macro:
+// 1) PID and TID are the same on single thread
+// 2) PID and TID are different in multi-thread
+TEST(gettid, MacroTest) {
+  EXPECT_EQ(getpid(), gettid());
+
+  pthread_t thread_id;
+  pid_t tid;
+  auto rc = pthread_create(&thread_id, NULL, ThreadedGettid, (void *)&tid);
+  if (rc) {
+    FAIL() << "Error: Unable to create thread.\n";
+  }
+
+  void *status;
+  rc = pthread_join(thread_id, &status);
+  if (rc) {
+    FAIL() << "Error: Unable to join thread.\n";
+  }
+
+  EXPECT_NE(getpid(), tid);
+}
+
+#define NUM_THREADS 10
```

Comment:
why not just const or constexpr?  no need to use define in C++ unless you need to be c compatible

Reply:
Removed in addition with the test that was removed.